tests/upgrade/upgrade_test.go

```diff
@@ -133,6 +134,54 @@ var _ = Describe("{UpgradeStork}", func() {
 	})
 })
 
+// UpgradeVolumeDriverReinstall test validates the scheduled app and re-install them. Making sure the operation and
+// new installation runs smoothly
+var _ = Describe("{UpgradeVolumeDriverReinstall}", func() {
+	var testrailID = 54370
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/54370
+	var runID int
+	JustBeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+
+	It("validate apps, upgrade volume driver, re-install apps, and ensure everything is running fine", func() {
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("b4upgradevolumedriverreinstall-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+
+		Step("start the upgrade of volume driver", func() {
+			err := Inst().V.UpgradeDriver(Inst().StorageDriverUpgradeEndpointURL,
+				Inst().StorageDriverUpgradeEndpointVersion,
+				Inst().EnableStorkUpgrade)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		Step("re-install and validate all apps after upgrade", func() {
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				contexts = append(contexts, ScheduleApplications(fmt.Sprintf("upgradevolumedriverreinstall-%d", i))...)
+			}
+
+			ValidateApplications(contexts)
+		})
+
+		Step("destroy apps", func() {
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+			for _, ctx := range contexts {
+				TearDownContext(ctx, opts)
+			}
```

Comment:
nit: opts can be specified inline on line 176:  `map[string]bool{"name": true}`

Reply:
updated.